### PR TITLE
Data transformation, telemetry, tests, and KSCrash source

### PR DESCRIPTION
**Pull request by benwong0**
@rokob included everything remaining for the iOS SDK here:

- Data transformation
- Telemetry data recording
- Unit tests
- Include KSCrash source so we don't need to include 2 frameworks

### CODE REVIEW

File: Rollbar/RollbarNotifier.m [hunk 1]
```diff
@@ -487,7 +623,21 @@ - (void)updateAccessToken:(NSString*)accessToken {
     self.configuration.accessToken = accessToken;
 }
 
-#pragma mark -
+#pragma mark - Network telemetry data
 
+- (void)captureTelemetryDataForNetwork:(BOOL)reachable {
```
Comment (rokob) [inline Rollbar/RollbarNotifier.m]: 👍

File: Rollbar/RollbarTelemetry.m [hunk 2]
```diff
@@ -0,0 +1,216 @@
+//
+//  RollbarTelemetry.m
+//  Rollbar
+//
+//  Created by Ben Wong on 11/21/17.
+//  Copyright © 2017 Rollbar. All rights reserved.
+//
+
```
Comment (rokob) [inline Rollbar/RollbarTelemetry.m]: This is not thread safe.

**Comment by rokob** [comment]
Rollbar defers concurrency issues to RollbarNotifier which uses RollbarThread to run the operation of actually sending items over the network. However, calls to `[Rollbar log:...]` concurrently from multiple threads appears to be racy. It looks like that is just how things currently are. So this looks like we are just continuing in that same threading model. Therefore the fact that RollbarTelemetry is not thread safe fits with the rest of this library.

Looks good to me.

**Comment by benwong0** [comment]
👍 sounds good. @rokob do you want me to merge or you will do it later?

**Pull request merged**